src/path/owned.rs

```diff
@@ -547,4 +597,25 @@ mod test {
             assert_eq!(path, expected.map(|x| x.to_owned()));
         }
     }
+
+    fn reparse_thing<T: std::fmt::Debug + std::fmt::Display + Eq + FromStr>(thing: T)
```

Comment:
does this need a `proptest` feature gate?

Reply:
No, because proptests are always enabled for testing.